lib/utilities/firebase_service.dart

```diff
@@ -40,17 +43,28 @@ class FirebaseService {
 
   // Register a user for an event
   Future<void> registerForEvent(String eventId, String userId) async {
-    try {
-      final eventDoc = _firestore.collection('events-collection').doc(eventId);
-      await eventDoc.update({
-        'registeredUsers': FieldValue.arrayUnion([userId]),
-        'slotsAvailable': FieldValue.increment(-1), // Decrement available slots
-      });
-      print('User $userId registered for event $eventId');
-    } catch (e) {
-      print('Error registering for event: $e');
-    }
+  try {
+    final eventDoc = _firestore.collection('events-collection').doc(eventId);
+
+    // Add user to the event's registered users list and decrement available slots
+    await eventDoc.update({
+      'registeredUsers': FieldValue.arrayUnion([userId]),
```

Comment:
We shouldn't track the registered users here. It's a duplicate from the registration collection. If, for some reason, the array of event.registedUsers is different than the data in the registration collection, then we would never know which one is right. You should only have one source of truth, not two.